include/libtorrent/upnp.hpp

```diff
@@ -240,8 +240,10 @@ struct TORRENT_EXTRA_EXPORT upnp final
 	void get_ip_address(rootdevice& d);
 	void delete_port_mapping(rootdevice& d, int i);
 	void create_port_mapping(http_connection& c, rootdevice& d, int i);
-	void post(upnp::rootdevice const& d, char const* soap
+	void post(upnp::rootdevice const& d, std::string const& soap
 		, char const* soap_action);
+	std::string create_soap(char const* soap_action
```

Comment:
all incoming strings here should probably be ``string_view`` too. Technically, returning a ``std::string`` here will be slightly more costly than the previous version, since it didn't heap allocate the string. not sure if anything need to be done about that though